### PR TITLE
cluster: Remove attributes meant only for RP-to-CS bridging

### PR DESCRIPTION
**Pull request by vkareh**

### CODE REVIEW

**Comment by vkareh** [comment]
@petrkotas @zgalor: I've removed a bunch of platform attributes that are either path of the URL path or things that CS will generate internally. Please take a look

File: api/redhatopenshift/Hypershift/hcpCluster-models.tsp [hunk 1]
```diff
@@ -96,10 +96,6 @@ model ClusterSpec {
   @visibility("create", "update")
   proxy?: ProxyProfile;
 
-  /** Additional trust bundle for the cluster */
-  @visibility("create", "update")
-  additionalTrustBundle?: string;
```
Comment (vkareh) [inline api/redhatopenshift/Hypershift/hcpCluster-models.tsp:101]: This is now part of the ProxyProfile (as `trustedCa`)

File: api/redhatopenshift/Hypershift/hcpCluster-models.tsp [hunk 2]
```diff
@@ -229,44 +225,26 @@ enum Visibility {
...
 
   /** The trusted CA for the proxy */
-  trustedCa: string;
+  trustedCa?: string;
 }
 
 /** Azure specific configuration */
 model PlatformProfile {
-  /** The Azure region for the cluster */
-  location: string;
```
Comment (vkareh) [inline api/redhatopenshift/Hypershift/hcpCluster-models.tsp:247]: Deployments are regional, we don't need to specify this here, right?

File: api/redhatopenshift/Hypershift/hcpCluster-models.tsp [hunk 3]
```diff
@@ -229,44 +225,26 @@ enum Visibility {
...
-
   /** Resource group to put cluster resources */
   managedResourceGroup: string;
 
-  /** vNet name that should be used for the cluster */
-  vnetName: string;
-
-  /** ResourceId for the subnet used by the control plane */
-  vnetId: string;
```
Comment (vkareh) [inline api/redhatopenshift/Hypershift/hcpCluster-models.tsp:256]: I think we create these in CS, so no need for exposing in API

File: api/redhatopenshift/Hypershift/hcpCluster-models.tsp [hunk 4]
```diff
@@ -229,44 +225,26 @@ enum Visibility {
...
-
   /** ResourceId for the subnet used by the control plane */
   subnetName: string;
 
-  /** Subscription ID */
-  subscriptionId: string;
```
Comment (vkareh) [inline api/redhatopenshift/Hypershift/hcpCluster-models.tsp:262]: This is part of the resource ID / URL path. We can set internally.

File: api/redhatopenshift/Hypershift/hcpCluster-models.tsp [hunk 5]
```diff
@@ -229,44 +225,26 @@ enum Visibility {
...
-
-  /** The identity id of the machine */
-  machineIdentityId: string;
-
-  /** The identity id of the control plane */
-  securityGroupName: string;
```
Comment (vkareh) [inline api/redhatopenshift/Hypershift/hcpCluster-models.tsp:268]: Do we own these or are they user configurable?

**Review by petrkotas — APPROVED** [review]

**Pull request merged**